src/libclipper/include/clipper/task_executor.hpp

```diff
@@ -101,9 +102,10 @@ class TaskExecutor {
             active_containers_->add_container(
                 vm, std::stoi(container_info["zmq_connection_id"]),
                 replica_id, parse_input_type(container_info["input_type"]));
-            // TODO: This callback should be submitted to a thread pool
-            // for execution on a separate thread
-            on_container_ready(vm, replica_id);
+
+            TaskExecutionThreadPool::submit_job([=]() {
```

Comment:
What is this closure capturing? Can we explicitly capture only the variables we need?